### PR TITLE
Reporting speed fixes

### PR DESCRIPTION
**Pull request by jsight**
The main change is that EffortReportModel now does a range index on effort.
The effort calculation queries now use this and run much quicker.
(also, hint and classification operations now show their timings in the detailed_stats.csv)

### CODE REVIEW

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-pr-builder/1234/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='windup-pr-builder/org.jboss.windup.rules.apps:windup-rules-java-tests' /><a href='http://wonka.mw.lab.eng.bos.redhat.com/jenkins/job/windup-pr-builder/1234/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport'>windup-pr-builder/org.jboss.windup.rules.apps:windup-rules-java-tests</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://wonka.mw.lab.eng.bos.redhat.com/jenkins/job/windup-pr-builder/1234/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport/org.jboss.windup.rules.apps.java.service/TypeReferenceServiceTest/testGetPackageUseFrequenciesSpeed/'><strong>org.jboss.windup.rules.apps.java.service.TypeReferenceServiceTest.testGetPackageUseFrequenciesSpeed</strong></a></li></ul>

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-core-PR-dependents-check/607/

**Comment by jsight** [comment]
retest

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-pr-builder/1235/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='windup-pr-builder/org.jboss.windup.rules.apps:windup-rules-java-tests' /><a href='http://wonka.mw.lab.eng.bos.redhat.com/jenkins/job/windup-pr-builder/1235/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport'>windup-pr-builder/org.jboss.windup.rules.apps:windup-rules-java-tests</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://wonka.mw.lab.eng.bos.redhat.com/jenkins/job/windup-pr-builder/1235/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport/org.jboss.windup.rules.apps.java.service/TypeReferenceServiceTest/testGetPackageUseFrequenciesSpeed/'><strong>org.jboss.windup.rules.apps.java.service.TypeReferenceServiceTest.testGetPackageUseFrequenciesSpeed</strong></a></li></ul>

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-core-PR-dependents-check/608/

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-core-PR-dependents-check/609/

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-pr-builder/1236/<h2>Failed Tests: <span class='status-failure'>2</span></h2><h3><a name='windup-pr-builder/org.jboss.windup.rexster:windup-rexster-tests' /><a href='http://wonka.mw.lab.eng.bos.redhat.com/jenkins/job/windup-pr-builder/1236/org.jboss.windup.rexster$windup-rexster-tests/testReport'>windup-pr-builder/org.jboss.windup.rexster:windup-rexster-tests</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://wonka.mw.lab.eng.bos.redhat.com/jenkins/job/windup-pr-builder/1236/org.jboss.windup.rexster$windup-rexster-tests/testReport/org.jboss.windup.graph.rexster.test.debug/RexsterDefaultDeploymentDebugTest/testRexsterProperStart/'><strong>org.jboss.windup.graph.rexster.test.debug.RexsterDefaultDeploymentDebugTest.testRexsterProperStart</strong></a></li></ul><h3><a name='windup-pr-builder/org.jboss.windup.rules.apps:windup-rules-java-tests' /><a href='http://wonka.mw.lab.eng.bos.redhat.com/jenkins/job/windup-pr-builder/1236/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport'>windup-pr-builder/org.jboss.windup.rules.apps:windup-rules-java-tests</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://wonka.mw.lab.eng.bos.redhat.com/jenkins/job/windup-pr-builder/1236/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport/org.jboss.windup.rules.apps.java.service/TypeReferenceServiceTest/testGetPackageUseFrequenciesSpeed/'><strong>org.jboss.windup.rules.apps.java.service.TypeReferenceServiceTest.testGetPackageUseFrequenciesSpeed</strong></a></li></ul>

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-core-PR-dependents-check/610/

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-pr-builder/1237/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='windup-pr-builder/org.jboss.windup.rules.apps:windup-rules-java-tests' /><a href='http://wonka.mw.lab.eng.bos.redhat.com/jenkins/job/windup-pr-builder/1237/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport'>windup-pr-builder/org.jboss.windup.rules.apps:windup-rules-java-tests</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://wonka.mw.lab.eng.bos.redhat.com/jenkins/job/windup-pr-builder/1237/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport/org.jboss.windup.rules.apps.java.service/TypeReferenceServiceTest/testGetPackageUseFrequenciesSpeed/'><strong>org.jboss.windup.rules.apps.java.service.TypeReferenceServiceTest.testGetPackageUseFrequenciesSpeed</strong></a></li></ul>

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-core-PR-dependents-check/611/

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-pr-builder/1238/

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-core-PR-dependents-check/612/

**Comment by mareknovotny** [comment]
As a whole it looks nice, I checked the performance and on my local run it is about 50 seconds faster in ReportRenderingPhase for phoenix app

**Comment by OndraZizka** [comment]
What's the main change(s)? It's a bit hard to get the idea from the changes alone, diff doesn't compare relevant parts... 
It seems like several minor optimizations.
If it works fine, it's okay to merge.

File: reporting/api/src/main/java/org/jboss/windup/reporting/model/EffortReportModel.java [hunk 1]
```diff
@@ -1,19 +1,24 @@
...
+@TypeValue(EffortReportModel.TYPE)
 public interface EffortReportModel extends WindupVertexFrame
 {
-    static final String TYPE = "ClassificationModel";
-    static final String TYPE_PREFIX = TYPE + ":";
-    static final String EFFORT = TYPE_PREFIX + "effort";
-    static final String SEVERITY = TYPE_PREFIX + TYPE_PREFIX + "severity";
+    String TYPE = "EffortReportModel";
+    String TYPE_PREFIX = TYPE + ":";
+    String EFFORT = "EffortReportModelEffort"; // don't use the prefix as we can't name the index with an "_"
```
Comment (OndraZizka) [inline reporting/api/src/main/java/org/jboss/windup/reporting/model/EffortReportModel.java]: Does the DB limit the name of the index?

File: reporting/api/src/main/java/org/jboss/windup/reporting/model/EffortReportModel.java [hunk 2]
```diff
@@ -1,19 +1,24 @@
...
+    String SEVERITY = TYPE_PREFIX + TYPE_PREFIX + "severity";
```
Comment (OndraZizka) [inline reporting/api/src/main/java/org/jboss/windup/reporting/model/EffortReportModel.java]: TYPE_PREFIX + TYPE_PREFIX ?

File: reporting/api/src/main/java/org/jboss/windup/reporting/model/EffortReportModel.java (hunk 1 above)
Comment (jsight) [inline reporting/api/src/main/java/org/jboss/windup/reporting/model/EffortReportModel.java]: Yes, it is a lucene limitation on search indexes.

**Pull request merged**